recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -187,4 +188,21 @@ this.RecipeRunner = {
       sandboxManager.evalInSandbox(actionScript);
     });
   },
+
+  /**
+   * Clear out cached state and fetch/execute recipes from the given
+   * API url. This is used mainly by the mock-recipe-server JS that is
+   * executed in the browser console.
+   */
+  testRun(baseApiUrl) {
+    const oldApiUrl = prefs.getCharPref("api_url");
+    prefs.setCharPref("api_url", baseApiUrl);
+
+    Storage.clearAllStorage();
+    NormandyApi.clearIndexCache();
+    this.start();
+
+    prefs.setCharPref("api_url", oldApiUrl);
```

Comment:
Can we wrap this up in a try/finally to make sure we reset the preferences and clear the cache, even if there is an error?